src/txkube/testing/integration.py

```diff
@@ -481,6 +487,74 @@ def check_configmaps(collection):
 
 
 
+class _PodTestsMixin(object):
+    @async
+    @needs(namespace=creatable_namespaces())
+    def test_pod(self, namespace):
+        """
+        ``Pod`` objects can be created and retrieved using the ``create`` and
+        ``list`` methods of ``IKubernetesClient``.
+        """
+        return self._create_list_test(
+            namespace, pods(), v1.Pod,
+            v1.PodList, matches_pod,
+        )
+
+
+    @async
+    @needs(namespace=creatable_namespaces())
+    def test_duplicate_pod_rejected(self, namespace):
+        """
+        ``IKubernetesClient.create`` returns a ``Deferred`` that fails with
+        ``KubernetesError`` if it is called with a ``Pod`` object with the
+        same name as a *Pod* which already exists in the same namespace.
```

Comment:
I guess I'm trying to differentiate between the Python object and the Kubernetes object.  Unsuccessfully, clearly. :)